### PR TITLE
typing: use autotyping to reduce implicit types

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

This PR runs [Autotyping](https://github.com/JelleZijlstra/autotyping) on all `nextcord/` files. This uses [LibCST's codemod](https://github.com/Instagram/LibCST) to refactor code to add explicit `-> None` returns, `__repr__` annotations, e.t.c.

This improves the `pyright --veriftypes` score by `0.8%`, which does not sound like much but is a lot for it being automatic.

## Checklist

- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/flags.py [hunk 1]
```diff
@@ -217,25 +217,25 @@ def _set_flag(self, o: int, toggle: bool) -> None:
             raise TypeError("Value to set for SystemChannelFlags must be a bool.")
 
     @flag_value
-    def join_notifications(self):
+    def join_notifications(self) -> int:
```
Comment (Skelmis) [inline nextcord/flags.py:220]: Aren't these technically wrong as ``@flag_value`` modifies them to return bools

Comment (ooliver1) [reply nextcord/flags.py:220]: Pyright recognises that.
image

Comment (Skelmis) [reply nextcord/flags.py:220]: Interesting, feel free to resolve then ig

**Review by teaishealthy — APPROVED** [review]
lgtm

**Review by Skelmis — APPROVED** [review]

**Comment by ooliver1** [comment]
libcst seems to have no wheels for 3.11, and pre-commit.ci does not allow customisation of the python version, not sure what to do.

**Pull request merged**